test/iterable_test.dart

```diff
@@ -609,9 +609,18 @@ void main() {
       );
     });
 
-    test('.asStream()', () {
-      expect([0, 1, 2, 3, 4, 5].asStream(), emitsInOrder([0, 1, 2, 3, 4, 5]));
-      expect([100, 99, 98, 95].asStream(), emitsInOrder([100, 99, 98, 95]));
+    group('.asStream()', () {
+      test('.asStream()', () {
+        expect([0, 1, 2, 3, 4, 5].asStream(), emitsInOrder([0, 1, 2, 3, 4, 5]));
+        expect([100, 99, 98, 95].asStream(), emitsInOrder([100, 99, 98, 95]));
+      });
+
+      test('.asStreamAwaited()', () {
+        expect(
+          [Future.value(0), Future.value(1)].asStreamAwaited(),
+          emitsInOrder([0, 1]),
```

Comment:
This test is weak. Replacing `asStreamAwaited()` with `asStream()` doesn't fail.

Reply:
Just checked the code and tested many implementations. All of them awaits for the result so I am afraid there is no way to provide a proper test for this case.
But I might be wrong. Can you please take a look?

Reply:
Done!